### DISCUSSION TITLE
Does Pode support the QUIC protocol (RFC 9000)?

### DISCUSSION THREAD

**Discussion by pcgeek86** [Badgerati/Pode#973] [Q&A] [answered]
Is there a way to configure a Pode endpoint (`Add-PodeEndpoint`) to explicitly use HTTP/2 or HTTP/3 over the UDP-based QUIC protocol?

* HTTP/2 can run over TCP
* HTTP/2 supports (but does not require) QUIC
* HTTP/3 requires QUIC

https://en.wikipedia.org/wiki/HTTP/3
https://en.wikipedia.org/wiki/QUIC

**Comment by Badgerati** [comment, marked as answer]
Hey @pcgeek86,

Pode at the moment doesn't support HTTP/2 or HTTP/3. It's been asked to have them (plus gRPC and QUIC) as a feature before here: #646.

It is something I want to investigate when I get chance.

**Reply by JustinGrote** [reply]
You'd have to compile probably for .NET 7 which supports HTTP/3 and HTTP/2 by default but that version would have to be Powershell 7.3 or higher only. Alternatively you could set the http version preference to the `OrHigher` or whatever its called and if you are on PowerShell 7.3 it should negotiate it.

**Reply by Badgerati** [reply]
Hi @JustinGrote,

For the main Pode module that might work if it relied on ASP.NET or HttpListener; instead Pode uses its own socket implementation, which means HTTP/2, 3, and gRPC support would each need coding into Pode.

However, there is the Pode.Kestrel module to use Kestrel as the listener - which definitely needs an upgrade at some point, and compiling for .NET7 - in which case using this might allow HTTP/2, 3 and gRPC support 🤔
